app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -0,0 +1,150 @@
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+
+        tabstrip.setupWithViewPager(viewpager)
+        hackToApplyTypefaces(tabstrip)
+
+        viewpager.adapter = viewPagerAdapter
+
+        tabstrip.addOnTabSelectedListener(TrackingOnTabSelectedListener(analytics, viewPagerAdapter))
+
+        setupToolbar()
+    }
+
+    private fun setupToolbar() {
+        toolbar.setTitle(R.string.activity_schedule)
+        toolbar.inflateMenu(R.menu.homepage)
+        toolbar.setOnMenuItemClickListener { item ->
+            when (item.itemId) {
+                R.id.action_search -> {
+                    navigate.toSearch()
+                    true
+                }
+                R.id.action_settings -> {
+                    navigate.toSettings()
+                    true
+                }
+                else -> false
+            }
+        }
+    }
+
+    override fun startLoading() {
+        subscriptions.add(
+                service.schedule(false)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(
+                            { updateWith(it, { event -> onEventClicked(event) }) },
+                            { Timber.e(it) }
+                    )
+        )
+    }
+
+    private fun onEventClicked(event: Event) {
+        analytics.trackItemSelected(ContentType.SCHEDULE_ITEM, event.id)
+        navigate.toEventDetails(event.id)
+    }
+
+    override fun stopLoading() {
+        subscriptions.dispose()
+    }
+
+    private fun hackToApplyTypefaces(tabLayout: TabLayout) {
+        // Unfortunately doing this the sensible way (in ScheduleViewPagerAdapter.getPageTitle())
+        // results in a bunch of other views on screen to stop drawing their text, for reasons only
+        // known to the Gods of Kobol. We can't theme things in the TabLayout either, as the
+        // TextAppearance is applied _after_ inflating the tab views, which means Calligraphy can't
+        // intercept that either. Sad panda.
+        tabLayout.addOnLayoutChangeListener { v, left, top, right, bottom, oldLeft, oldTop, oldRight, oldBottom ->
+            val context = tabLayout.context
+            val typeface = context.getFontFor(R.style.TextAppearance_Squanchy_Tab)
+
+            val tabCount = tabLayout.tabCount
+            for (i in 0 until tabCount) {
+                val tab = tabLayout.getTabAt(i)
+                if (tab == null || hasTypefaceSpan(tab.text)) {
+                    continue
+                }
+                tab.text = tab.text?.applyTypeface(typeface)
+            }
+        }
+    }
+
+    private fun hasTypefaceSpan(text: CharSequence?) = if (text !is Spanned) {
+        false
+    } else {
+        text.hasTypefaceSpan()
+    }
+
+    fun updateWith(schedule: Schedule, onEventClicked: (Event) -> Unit) {
+        viewPagerAdapter.updateWith(
+                schedule.pages, object : ScheduleViewPagerAdapter.OnEventClickedListener {
```

Comment:
I think you need to carry through the `(Event) -> Unit` into here too, it's missing the second half of the change otherwise

Reply:
Done in 404dc27061426462b698b82e8271ddd40aa06750
I had to change some other Java classes, which will be converted in another PR to be compatible with the change